go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/grafana/azure-data-explorer-datasource
 go 1.17
 
 require (
+	github.com/AzureAD/microsoft-authentication-library-for-go v0.4.0
```

Comment:
Seems like this lib is in production preview. Guess it's fine then? Is there a reason this is not part of the azure-sdk-go?

Reply:
`azure-sdk-go` does actually use it for client secret scenario, same version of MSAL library (indirect dependency changes to direct dependency).
They had their own implementation in past but replaced it by MSAL: [here](https://github.com/Azure/azure-sdk-for-go/blob/a516802508c2239b7782b8c3fd26b6d582357e21/sdk/azidentity/client_secret_credential.go#L48) and [here](https://github.com/Azure/azure-sdk-for-go/blob/a516802508c2239b7782b8c3fd26b6d582357e21/sdk/azidentity/client_secret_credential.go#L65)

Usage of MSAL in my `aad_client.go` is very similar to how they use it in their `client_secret_credential.go`.
Azure SDK doesn't support OBO scenario, so I decided to refence MSAL directly, similar how they do it for client secret.

Reply:
Okay, thanks for making this clear!